### PR TITLE
set wider permissions on log directory

### PR DESCRIPTION
**Pull request by mwhudson**

### CODE REVIEW

File: subiquitycore/log.py [hunk 1]
```diff
@@ -13,12 +13,16 @@
...
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+import grp
 import logging
 import os
 
 
 def setup_logger(dir, base='subiquity'):
-    os.makedirs(dir, exist_ok=True)
+    os.makedirs(os.path.basename(dir), exist_ok=True)
```
Comment (seb128) [inline subiquitycore/log.py:22]: Why this change? It seems not needed for the permission and probably wrong (or it relies on the helper to run in the right location?)

File: subiquitycore/log.py [hunk 2]
```diff
@@ -13,12 +13,16 @@
...
+    if os.getuid() == 0:
+        os.chmod(dir, 0o775)
+        os.chown(dir, 0, grp.getgrnam('adm').gr_gid)
```
Comment (seb128) [inline subiquitycore/log.py:25]: the documentation states 'To leave one of the ids unchanged, set it to -1', wouldn't it be better than forcing 0 as uid?

**Review by seb128 — COMMENTED** [review]
Thanks. It should allow for the graphical installer to write to the log but only after the subiquity service is started, it still feel like to me that it would be more robust to create the dir with the right permission on the iso or in casper.

File: subiquitycore/log.py (hunk 1 above)
Comment (mwhudson) [reply subiquitycore/log.py:22]: Because I shouldn't write code late at night :) (I had a version that passed a mode to mkdir but that ran into issues with umask)

File: subiquitycore/log.py (hunk 2 above)
Comment (mwhudson) [reply subiquitycore/log.py:25]: Ah TIL.

**Review by seb128 — APPROVED** [review]
Thanks, it should help at least in cases where the service is active before the frontend tries to create the log
